core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -550,37 +555,6 @@ class AvroUpsertMonitorActor(val pollInterval: FiniteDuration,
     pubSubDao.acknowledgeMessagesById(pubSubSubscriptionName, scala.collection.immutable.Seq(ackId))
   }
 
-  case class AvroUpsertAttributes(workspace: WorkspaceName,
-                                  userEmail: RawlsUserEmail,
-                                  importId: UUID,
-                                  upsertFile: String,
-                                  isUpsert: Boolean
-  )
-
-  private def parseMessage(message: PubSubMessage) = {
-    val workspaceNamespace = "workspaceNamespace"
-    val workspaceName = "workspaceName"
-    val userEmail = "userEmail"
-    val jobId = "jobId"
-    val upsertFile = "upsertFile"
-    val isUpsert = "isUpsert"
-
-    def attributeNotFoundException(attribute: String) = throw new Exception(
-      s"unable to parse message - attribute $attribute not found in ${message.attributes}"
-    )
-
-    AvroUpsertAttributes(
-      WorkspaceName(
-        message.attributes.getOrElse(workspaceNamespace, attributeNotFoundException(workspaceNamespace)),
-        message.attributes.getOrElse(workspaceName, attributeNotFoundException(workspaceName))
-      ),
-      RawlsUserEmail(message.attributes.getOrElse(userEmail, attributeNotFoundException(userEmail))),
-      UUID.fromString(message.attributes.getOrElse(jobId, attributeNotFoundException(jobId))),
-      message.attributes.getOrElse(upsertFile, attributeNotFoundException(upsertFile)),
-      java.lang.Boolean.parseBoolean(message.attributes.getOrElse(isUpsert, "true"))
-    )
-  }
-
```

Comment:
The `parseMessage` function has moved to `AvroUpsertMonitorMessageParser`, just makes it a bit more encapsulated and easy to test